events/events_network.c

```diff
@@ -136,6 +136,15 @@ events_network_register(int (*func)(void *), void * cookie, int s, int op)
 		goto err0;
 
 	/*
+	 * There is a new socket/operation pair:
+	 * Rebuild the fd list on the next call to events_network_select().
+	 */
+	if (fds != NULL) {
```

Comment:
Taking a bit of time is fine.  This issue has sat open for quite a while already. :-)

But I wonder if it would be enough to set a flag for "regenerate list" rather than throwing it away entirely here...?